[F11-1] iScheduler Xs Max

HEARTOFAL1ON opened the pull request:
@YezhongZ @Hexinyiyi @ChengYuuu @jiacheng-panda

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -56,18 +56,23 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+* `add n/Study With Yoko s/tomorrow at 3pm d/Study again and again p/high rs/MONTHLY ru/ next friday 3pm tag/ad-hoc`
+* `add n/CS2103 Lecture s/23 September 2018 6pm e/ 24 September 2018 p/low rs/WEEKLY ru/ next monday tag/timetable`
 
 === Listing all persons : `list`
```
I think this list command should be updated to match the `list` command in the examples above at line 36

liliwei25 commented on src/main/java/seedu/address/logic/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,69 @@
+public class AddEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "add";
+
```
Command alias seems to be missing

liliwei25 commented on src/main/java/seedu/address/model/event/DateTime.java:
```diff
@@ -0,0 +1,75 @@
+     *
+     * @param formmatedDateTime A local date time.
+     */
+    public DateTime(LocalDateTime formmatedDateTime) {
```
typo error in 'frommatedDateTime'

HEARTOFAL1ON replied:
done

liliwei25 commented on src/main/java/seedu/address/model/event/DateTime.java:
```diff
@@ -0,0 +1,75 @@
+
+    /**
+     * Used for PopUp to check if the current time is close to the next event in the pop up queue
+     * @param currentDateTime
```
There should be a blank line before '@param'

HEARTOFAL1ON replied:
done

liliwei25 commented on src/test/java/seedu/address/model/ModelManagerTest.java:
```diff
@@ -1,10 +1,17 @@
 package seedu.address.model;
 
+//import static org.junit.Assert.assertEquals;
```
Remove unused imports

liliwei25 commented on src/test/java/seedu/address/logic/commands/AddEventCommandTest.java:
```diff
@@ -0,0 +1,229 @@
+        new AddEventCommand(null);
+    }
+
+    /* @Test still failing
```
Try not to leave commented code in, even if it is to pass the tests. Either fix the tests or remove them temporarily.

liliwei25 commented on src/main/java/seedu/address/model/event/EventName.java:
```diff
@@ -0,0 +1,59 @@
+    /**
+     * Returns true if a given string is a valid event name.
+     */
+    public static boolean isValidEventName(String test) {
```
'test' is not a good variable/parameter name

HEARTOFAL1ON replied:
done

liliwei25 reviewed (commented):
No major issues. Good job! 👍

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -242,8 +247,8 @@ _{explain how the user can enable/disable data encryption}_
-* *Add* `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` +
-e.g. `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+* *Add event* `add event n/EVENT_NAME [s/EVENT_START_DATE_TIME] [e/EVENT_END_DATE_TIME] [d/DESCRIPTION] [p/PRIORIY] [v/VENUE] [rs/EVENT_RECURRING_STATUS] [ru/EVENT_RECURRING_UNTIL_DATE_TIME] [t/TAG]…​` +
+e.g. `add n/Study With Yoko s/tomorrow at 3pm d/Study again and again p/high r/MONTHLY tag/ad-hoc`
```
Should this be t/ or tag/ ?

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -56,18 +56,23 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 
-* `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`
-* `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal`
+* `add n/Study With Yoko s/tomorrow at 3pm d/Study again and again p/high rs/MONTHLY ru/ next friday 3pm tag/ad-hoc`
```
t/ or tag/?

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -316,14 +316,74 @@ image::UndoRedoActivityDiagram.png[width="650"]
+**Step 2** The user executes `GetGCEvents` command to download the Google Calendar Events to local database. +
+An Calendar object is to be built with method `getCalendar`. +
+A method `getNetHttpTransport` is called to let application establish a connection with Google to initiate an authorization process
+(for the use who use this command for the  first time).
```
typo here?

YipZong replied:
Will fix in next PR.

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -316,14 +316,74 @@ image::UndoRedoActivityDiagram.png[width="650"]
+(for the use who use this command for the  first time).
+
+A pop-up window will appear. The User will be required to log in their Google Account as per other Google Service. +
+After successful login, the login credential will be saved locally such that no future log-in is required. +
```
Is the credentials saved securely?

YipZong replied:
Yes :) It is not account/password. The stored info cannot be used to calculate backwards for the sensitive infoamtion.

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -316,14 +316,74 @@ image::UndoRedoActivityDiagram.png[width="650"]
+For current implementation, only events in the primary (default) are extracted.
+====
+
+**Step 5** For each event extracted, a [proposed] eventConvertor utility method called `addGcEventToLocal` is called to convert the Google Extreacted Event to local Event.
```
typo here

YipZong replied:
Will fix in the PR soon.

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -351,6 +411,153 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+==== Current Implementation
+
+The `add` command allows user to add new events to the scheduler. Add Command is facilitated by the `AddCommand` class.
+The format of this command is `add event n/EVENT_NAME [s/START_DATETIME] [e/END_DATETIME] [d/DESCRIPTION] [v/VENUE] [rt/RECURRING_TYPE] [ru/RECURRING_UNTIL_DATETIME] [t/TAG]…​`.
```
So users can add events without the start and end date/time?

liliwei25 commented on _reposense/config.json:
```diff
@@ -0,0 +1,30 @@
+      "githubId": "HEARTOFAL1ON",
+      "displayName": "DANIE...E YAO",
+      "authorNames": ["HEARTOFAL1ON"]
+    }
```
add a comma after each author

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -1017,3 +1224,38 @@ _{ more test cases ... }_
+
+*Zhang Yezhong*
+
+v1.1
```
Please update accordingly

liliwei25 commented on src/main/java/seedu/scheduler/MainApp.java:
```diff
@@ -61,17 +56,16 @@
     @Override
     public void init() throws Exception {
-        logger.info("=============================[ Initializing AddressBook ]===========================");
+        logger.info("=============================[ Initializing Scheduler ]====================+=======");
```
typo

liliwei25 commented on src/test/java/seedu/scheduler/ui/EventListPanelTest.java:
```diff
@@ -0,0 +1,129 @@
+     * {@code CARD_CREATION_AND_DELETION_TIMEOUT} milliseconds to execute.
+     */
+    /**
+    @Test
```
commented code?

liliwei25 commented on src/test/java/seedu/scheduler/ui/BrowserPanelTest.java:
```diff
@@ -0,0 +1,41 @@
+    }
+    /**
+    @Test
+    public void display() throws Exception {
```
commented code?

liliwei25 commented on src/test/java/seedu/scheduler/testutil/EventUtil.java:
```diff
@@ -0,0 +1,83 @@
+        descriptor.getRepeatUntilDateTime().ifPresent(repeatUntilDateTime -> sb.append(PREFIX_REPEAT_UNTIL_DATE_TIME)
+                .append(repeatUntilDateTime.getPrettyString()).append(" "));
+        sb.append(descriptor.getReminderDurationList().get().getPrettyString()).append(" ");
+        /*System.out.println(descriptor.getReminderDurationList().get().get());
```
commented code

liliwei25 commented on src/main/java/seedu/scheduler/model/event/ReminderTimeList.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @return the current number of reminder times
+     */
+    public int size() {
```
check naming convention

liliwei25 commented on src/main/java/seedu/scheduler/model/event/ReminderDurationList.java:
```diff
@@ -0,0 +1,120 @@
+     *
+     * @return the current number of reminder times
+     */
+    public int size() {
```
check naming convention

liliwei25 commented on src/main/java/seedu/scheduler/model/event/ReminderDurationList.java:
```diff
@@ -0,0 +1,120 @@
+        }
+    }
+
+    public Map<Duration, Boolean> get () {
```
get what?

liliwei25 commented on src/main/java/seedu/scheduler/model/Model.java:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Returns true if a tag with the same identity as {@code tag} exists in the scheduler.
+     */
+    //boolean hasTag(Tag tag);
```
commented code

liliwei25 reviewed (commented):
Good job! 👍 Just some minor issues.
* AddCommandSequenceDiagram: Some lifelines are missing or missing object deletion notation (X)
* What are the Background.png files for?

ghost commented:
Hi, these PRs may be of interest: [PR 140](https://github.com/CS2103-AY1819S1-F11-1/main/pull/140), 
[PR 146](https://github.com/CS2103-AY1819S1-F11-1/main/pull/146) and [PR 147](https://github.com/CS2103-AY1819S1-F11-1/main/pull/147).

They are the make-up for the DG portion for milestone v1.2. Apologies for the delay~

ghost commented:
Hi, for this [PR](https://github.com/CS2103-AY1819S1-F11-1/main/pull/156), please also refer to my PR description :). It gives acknowledgments as per module policy (regardless whether significant or insignificant, I guess that acknowledgments must be given)

https://github.com/CS2103-AY1819S1-F11-1/main/pull/156#issue-227532069

YipZong committed:
Add logger infomation

YipZong committed:
update logger text

zylonc committed:
Add more Tests

Added more test for
1. EventList
2. RepeatEventGenerator
3. ParserUtil

YipZong committed:
Extract a method for updateSingleNonRepeatEvent, restructure logic

YipZong committed:
Add a method to tell whether the event is a repeat event

YipZong committed:
Condense method updateSingleGoogleEvent

YipZong committed:
Add a static method get EventUid and EventSetUid In Google Format From a Local Event

YipZong committed:
replace magic text "primary"

YipZong committed:
Simpify code, fix code quality

YipZong committed:
Delete unnecessary code

YipZong committed:
Update updateAllGoogleEvent

YipZong committed:
Combine editUpcoming and editAll

YipZong committed:
Fix counter name

YipZong committed:
Condense getRecurringEventId() method

YipZong committed:
Update updateSingleRepeatInstance

zylonc committed:
Merge pull request #172 from YezhongZ/condenseEditCommand

EditCommand: condense code

YipZong committed:
add test ulti

YipZong committed:
Add test for add repeated events

YipZong committed:
Add test for edit repeated events

YipZong committed:
fix style

liliwei25 commented on src/main/java/seedu/scheduler/MainApp.java:
```diff
@@ -207,6 +211,7 @@ public void start(Stage primaryStage) {
     @Override
     public void stop() {
         logger.info("============================ [ Stopping Scheduler ] =============================");
+        // model.syncWithPopUpManager(PopUpManager.getInstance(), storage);
```
remove unwanted code

liliwei25 commented on src/main/java/seedu/scheduler/commons/web/ConnectToGoogleCalendar.java:
```diff
@@ -0,0 +1,735 @@
+            events = service.events().list(CALENDAR_NAME)//set the source calendar on google
+                    .setMaxResults(numberOfEventsToBeDownloaded) //set upper limit for number of events
+                    .setTimeMin(now)//set the starting time
+                    //.setOrderBy("startTime")//if not specified, stable order
```
remove unused code

liliwei25 commented on src/main/java/seedu/scheduler/logic/RepeatEventGenerator.java:
```diff
@@ -0,0 +1,186 @@
+     * Generate all events that are repeated daily from {@code targetEvent}.
+     * Returns a list of events that are repeated daily.
+     */
+    public List<Event> generateDailyRepeatEvents(Event targetEvent) {
```
these `generateXXXRepeatEvents` contain a lot of repeated code. is it possible to reuse some of these?

liliwei25 commented on src/test/java/systemtests/SchedulerSystemTest.java:
```diff
@@ -216,11 +216,12 @@ protected void assertSelectedCardChanged(Index expectedSelectedCardIndex) {
         }
-        assertEquals(expectedUrl, getBrowserPanel().getLoadedUrl());
+        // Remove for v1.3.1, because of the readme page
+        // assertEquals(expectedUrl, getBrowserPanel().getLoadedUrl());
```
remove unwanted code

liliwei25 commented on src/test/java/systemtests/DeleteCommandSystemTest.java:
```diff
@@ -59,31 +59,33 @@ public void delete() {
         int invalidIndex = getModel().getScheduler().getEventList().size();
         command = DeleteCommand.COMMAND_WORD + " " + invalidIndex;
         assertCommandFailure(command, MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+        //remove for v1.3.1 for the readme page will change thr URL
```
remove unwanted code

liliwei25 commented on src/main/java/seedu/scheduler/model/event/EventPopUpInfo.java:
```diff
@@ -1,30 +1,124 @@
+
+    /**
+     * get Display for events
+     * @return
```
missing comment

liliwei25 commented on src/main/java/seedu/scheduler/model/event/EventPopUpInfo.java:
```diff
@@ -1,30 +1,124 @@
-        return description;
+    /**
+     * get Display for events that are past
+     * @return
```
missing comment

liliwei25 commented on src/main/java/seedu/scheduler/model/PopUpManager.java:
```diff
@@ -20,53 +29,298 @@
 
                     if (!popUpQueue.isEmpty()) {
-                        logger.info("Checking for incoming events");
+                        //logger.info("Checking for incoming events");
```
remove unwanted code

liliwei25 commented on src/main/java/seedu/scheduler/model/PopUpManager.java:
```diff
@@ -20,53 +29,298 @@
+                    if (!flag) {
+                        if (!popUpQueue.isEmpty()) {
+                            DateTime frontEventDateTime = popUpQueue.peek().getPopUpDateTime();
+                            //logger.info(frontEventDateTime.toString());
```
remove unwanted code

liliwei25 commented on src/main/java/seedu/scheduler/model/PopUpManager.java:
```diff
@@ -20,53 +29,298 @@
+
+    /**
+     * Util function that returns now in DateTime format
+     * @return
```
missing comment

liliwei25 commented on src/main/java/seedu/scheduler/model/PopUpManager.java:
```diff
@@ -20,53 +29,298 @@
+     * Check if eventPopUpInfo belongs to an upcoming event compared to target
+     * @param event
+     * @param target
+     * @return
```
missing comment for this and the functions below

liliwei25 commented on src/main/java/seedu/scheduler/model/PopUpManager.java:
```diff
@@ -20,53 +29,298 @@
+    private static PopUpManager instance;
     private PriorityQueue<EventPopUpInfo> popUpQueue;
+    private Boolean flag;
+    // private static ArrayList<EventPopUpInfo> pastPopUps;
```
remove unwanted code

liliwei25 commented on src/main/java/seedu/scheduler/logic/parser/EditCommandParser.java:
```diff
@@ -48,6 +50,23 @@ public EditCommand parse(String args) throws ParseException {
+        }
+
+
+        /*
```
remove unwanted code

liliwei25 commented on src/main/java/seedu/scheduler/logic/commands/EditCommand.java:
```diff
@@ -47,20 +59,25 @@
 
     /**
-     * @param index of the event in the filtered event list to edit
+     * @param index               of the event in the filtered event list to edit
```
extra spaces?

liliwei25 reviewed (commented):
* For ModelClassBetterOopDiagram:
  * please do not use X, just remove the unwanted parts of the diagram
* For ModelClassDiagram:
  * please do not use X, just remove the unwanted parts of the diagram
* Overall is okay except there are quite a number of unwanted codes, missing comments and extra blank lines.

Good Job 👍  Good luck for your PE! 🙂

YipZong committed:
Update typo

YipZong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main into allowOfflineMode

YipZong committed:
update return to bool instead of void to indicate result

YipZong committed:
Update GetGCEvents

YipZong committed:
Update addCommand: support offline mode

YipZong committed:
Fix type

YipZong committed:
update logic to prevent second getGCEvents

YipZong committed:
supply disable token

YipZong committed:
Add newline at EOF

zylonc committed:
Implement real time date time parser

Date time entered in commands will be parsed realtime and display in
result display box.

zylonc committed:
Merge pull request #171 from ChengYuuu/enhance_test

Add Test cases

zylonc committed:
Merge pull request #222 from ChengYuuu/ui_parser

Implement real time date time parser

zylonc committed:
RepeatEventGenerator.java: Change repeat logic

Repeated events x will always fall in the range of
1st start event start date time < x < event's repeat until date time

zylonc committed:
EditCommand.java: Handle Event DateTime Issues

Events edited with start date time later than end date time will throw an error message when it is being edited.

zylonc committed:
Refactor Repeat Event Generator

Refactored out Repeat Event Generator and also add a check for repeat
until date time and end date time.

YipZong committed:
DeleteCommand: add command status

YipZong committed:
EditCommand: add command status

YipZong committed:
Merge branch 'master' into allowOfflineMode

YipZong committed:
fix style

YipZong committed:
add new library for BiMap

YipZong committed:
add new library for BiMap

YipZong committed:
fix getGCEvents

zylonc committed:
Update PPP and UserGuide

zylonc committed:
DeveloperGuide.adoc: Update use cases

YipZong committed:
Update doc

YipZong committed:
Merge pull request #233 from ChengYuuu/update_ppp

Update PPP and UserGuide

YipZong committed:
Merge pull request #232 from ChengYuuu/update_use_case

DeveloperGuide.adoc: Update use cases

YipZong committed:
Merge pull request #225 from ChengYuuu/revamp_edit

RepeatEventGenerator.java: Change repeat logic

zylonc committed:
Merge pull request #234 from YezhongZ/UpdateDoc

Update doc

YipZong committed:
fix style, add logger

YipZong committed:
fix code quality

YipZong committed:
Merge branch 'master' into AddTest

YipZong committed:
change variable access definition order

Hexinyiyi committed:
Merge pull request #216 from YezhongZ/AddTest

Add Tests for add and edit

Hexinyiyi committed:
1. Fix pop up manager bug (only update when reminderDurationList changes)
2. Implement AddReminderCommand and its parser

Hexinyiyi committed:
Merge pull request #230 from YezhongZ/FixGetGCEvents

Fix GetGCEvents

YipZong committed:
Merge branch 'master' into allowOfflineMode

YipZong committed:
fix typo

YipZong committed:
Merge pull request #220 from YezhongZ/allowOfflineMode

Allow offline mode

zylonc committed:
Merge pull request #228 from ChengYuuu/refactor_repeat_event_generator

Refactor Repeat Event Generator

YipZong committed:
fix delete single event: previously report internet error although already deleted

Hexinyiyi committed:
Implement AddReminder and DeleteReminder command

Hexinyiyi committed:
Merge pull request #239 from YezhongZ/fixDeleteSingleEvent

fix delete single event

zylonc committed:
Merge branch 'master' into edit_date_restrict

YipZong committed:
fix comp method

YipZong committed:
Merge pull request #226 from ChengYuuu/edit_date_restrict

EditCommand.java: Handle Event DateTime Issues

zylonc committed:
Merge pull request #244 from YezhongZ/fixRepeatEventRelated

fix instance comparison method

YipZong committed:
Enhance offlinemode: now by default everything is offline unless initialized to online mode

YipZong committed:
Refactor command name to EnterGoogleCalendarMode

YipZong committed:
fix code quality

YipZong committed:
refactor command name to EnterGoogleCalendarModeCommand

zylonc committed:
Fix Event Constructor(event uid)

YipZong committed:
Merge pull request #253 from ChengYuuu/fix_add

Fix Event Constructor(event uid)

YipZong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main into enhanceOfflineMode

YipZong committed:
Fix the first few instance disappear problem

YipZong committed:
fix style

YipZong committed:
Add test case

YipZong committed:
improve code quality

YipZong committed:
fix style

Hexinyiyi committed:
Merge pull request #252 from YezhongZ/enhanceOfflineMode

Enhance offline mode

Hexinyiyi committed:
Merge pull request #257 from YezhongZ/FixEdit-uSecondInstance

Fix the first few instance disappear problem

YipZong committed:
enhance clear command to support google cal

voltaireesquirez committed:
Remove instances of "address", "address book", "person" in project

voltaireesquirez committed:
.gitignore: Ignore tokens/mode.txt

voltaireesquirez committed:
Update version of app to v1.4

Hexinyiyi committed:
Implement PostponeReminder Function

Hexinyiyi committed:
Push postponeReminderCommandParser

voltaireesquirez committed:
Change MESSAGE_USAGE for Edit command

Hexinyiyi committed:
Make the reminderlist display in order

Hexinyiyi committed:
Revert back delay time

YipZong committed:
add test, edit text

Hexinyiyi committed:
Merge branch 'org'

YipZong committed:
fix style

YipZong committed:
update test

YipZong committed:
update test

YipZong committed:
remove unused variable

Hexinyiyi committed:
Merge pull request #259 from YezhongZ/EnhanceClearCommand

enhance clear command to support google cal

YipZong committed:
Add test case, supply credential with write permission

YipZong committed:
add test for setGoogleCalendarStatus

Hexinyiyi committed:
Merge pull request #263 from YezhongZ/AddNewTestCase

Add test case, supply credential with write permission

Hexinyiyi committed:
Fix coding style

Hexinyiyi committed:
Merge branch 'org'

# Conflicts:
#	tokens/StoredCredential

YipZong committed:
Change method signature, add test

Hexinyiyi committed:
Block editing recurring events for Add/Delete/Postpone Reminders

Hexinyiyi committed:
Delete BrowserPanel.java

zylonc committed:
Fix Event Constructor for eventUid

Hexinyiyi committed:
Revert back delay time

Hexinyiyi committed:
merge remote branch master to local

Hexinyiyi committed:
Change delay time

voltaireesquirez committed:
Merge pull request #262 from HEARTOFAL1ON/v1.4_milestone

final milestone

YipZong committed:
Merge pull request #266 from Hexinyiyi/master

Add add/delete/postpone Reminder features

YipZong committed:
add new test method

YipZong committed:
add new typical event

YipZong committed:
add new test

YipZong committed:
remove un-implemented methods

YipZong committed:
addEvent test, fix google get cal test (got limit)

YipZong committed:
combined deleteUpcoming and deleteAll

YipZong committed:
improve command

YipZong committed:
fix test

YipZong committed:
fix test

YipZong committed:
Merge pull request #267 from ChengYuuu/event_constructor

Fix Event Constructor for eventUid

YipZong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main into AddNewTestCase

voltaireesquirez committed:
docs/team/heartofal1on.adoc: Update PPP

YipZong committed:
re-factor

YipZong committed:
add helper command

YipZong committed:
test delete

Hexinyiyi committed:
add addReminderCommand test cases. Add parse duration regex

zylonc committed:
Merge pull request #269 from CS2103-AY1819S1-F11-1/updatePPP_dan

Update PPP for v1.4

YipZong committed:
Update test for add and delete single

YipZong committed:
fix style

Hexinyiyi committed:
Allow reminder commands follow the same logic as edit, fix bugs in popUpManager

YipZong committed:
fix bug: delete single event causes error

zylonc committed:
Change Edit Event Logic

1. Single Event:
Repeat Until Date Time will be pegged to end date time
Edit (Event) e/tomorrow, ru/ will be tomorrow also

This is okay because we cannot edit ru/ for single events.

So, single event end date time is not constrained by repeat until date
time

2. Repeated Event:
Repeat until date time will be considered seperately from event end date
time.

This means Repeat until date time will be restricting the event end date
time.

Closes #254

Added tests for editing upcoming and all repeat events.

Hexinyiyi committed:
Change themes

zylonc committed:
Restrict users from using options for single event

Throws an command exception when user tries to input options such as -a
or -u for single events

YipZong committed:
test for repeat

Hexinyiyi committed:
Correct .equal method for Event and write test cases

Hexinyiyi committed:
Merge remote-tracking branch 'orimas/master'

# Conflicts:
#	src/main/resources/view/DarkTheme.css

YipZong committed:
Update test for delete

voltaireesquirez committed:
docs/team/heartofal1on.adoc: Update PPP

YipZong committed:
add new event for testing

YipZong committed:
refactor delete

Hexinyiyi committed:
Complete tests for AddReminderCommand

Hexinyiyi committed:
1. Delete unused syncWithPopUpManager
2. Complete test for DeleteReminderCommand

YipZong committed:
Add edit Google Events tests

YipZong committed:
fix style

Hexinyiyi committed:
Fix cases where users enter empty reminder

YipZong committed:
More test for delete, add one more test to cover reminder

Hexinyiyi committed:
Complete test for PostponeReminderCommand

zylonc committed:
DeveloperGuide.adoc: Update edit implementation

Hexinyiyi committed:
Complete test for Add/DeleteReminderCommandParser

Hexinyiyi committed:
Complete test for PostponeReminderCommandParser

zylonc committed:
Merge pull request #271 from CS2103-AY1819S1-F11-1/updatePPP_dan

Update PPP

Hexinyiyi committed:
Correct coding styles for all tests

Hexinyiyi committed:
Merge branch 'org'

zylonc committed:
DeveloperGuide.adoc: Change add implementation

Hexinyiyi committed:
Delete unused function defined

Hexinyiyi committed:
Change gitignore

Hexinyiyi committed:
Revert "Change gitignore"

This reverts commit 194786954be290e7c2d87f5ddd8e71c576ca2246.

Hexinyiyi committed:
Add Test for EventPopUpInfo

Hexinyiyi committed:
Change AddCommandSystemTest

YipZong committed:
add test for EnterGoogleCaMode

Hexinyiyi committed:
Comment out unused import first

YipZong committed:
fix style

zylonc committed:
Merge pull request #272 from Hexinyiyi/master

Fix ReminderCommands Logic and write test cases

Hexinyiyi committed:
Add test for DateTime Methods and EventPopUpInfo

YipZong committed:
try

YipZong committed:
Merge branch 'master' into AddNewTestCase

YipZong committed:
fix style

YipZong committed:
fix test

Hexinyiyi committed:
Write tests for PopUpManager

YipZong committed:
fix style

YipZong committed:
try to fix test

zylonc committed:
Merge branch 'master' into edit_change

YipZong committed:
Improve test, fix style

zylonc committed:
Merge pull request #265 from YezhongZ/AddNewTestCase

Add test

YipZong committed:
merge

YipZong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main

YipZong committed:
Refactor EnterGoogleCalendarModeCommand

YipZong committed:
fix code quality

YipZong committed:
add logger

zylonc committed:
Merge branch 'master' into edit_change

zylonc committed:
Merge pull request #270 from ChengYuuu/edit_change

Change Edit Event Logic

YipZong committed:
refactor connectToGoogleCalendar

Hexinyiyi committed:
Fix typical event scheduler uid and passes all existing tests

Hexinyiyi committed:
Complete tests for PopUpManager

Hexinyiyi committed:
conclude merging with local and remote org master branch

YipZong committed:
refactor EventFormatUntil

YipZong committed:
refactor the location of class

YipZong committed:
remove clear Google test

YipZong committed:
fix style

Hexinyiyi committed:
Merge all sample events and fix coding style

Hexinyiyi committed:
Merge pull request #278 from YezhongZ/removeClearGoogleTest

Remove clear google test

YipZong committed:
Revert "Remove clear google test"

Hexinyiyi committed:
Merge pull request #281 from CS2103-AY1819S1-F11-1/revert-278-removeClearGoogleTest

Revert "Remove clear google test"

Hexinyiyi committed:
Merge pull request #276 from YezhongZ/Refactor

refactor EnterGoogleCalendarModeCommand

YipZong committed:
Merge pull request #280 from Hexinyiyi/master

Add tests for PopUpManager

YipZong committed:
Remove Clear Google Cal test

YipZong committed:
update icon

Hexinyiyi committed:
Merge pull request #282 from YezhongZ/RemoveClearGoogleCal

Remove Clear Google Cal test

YipZong committed:
Merge pull request #283 from YezhongZ/UpdateIcon

update icon

YipZong committed:
Update Ui

Hexinyiyi committed:
Update userguide

Hexinyiyi committed:
Merge branch 'org'

YipZong committed:
Merge pull request #285 from Hexinyiyi/master

Update userguide

Hexinyiyi committed:
Merge pull request #284 from YezhongZ/updateUi

Update Ui

YipZong committed:
refactor

YipZong committed:
fix style

YipZong committed:
fix style add doc

voltaireesquirez committed:
Final changes to README

voltaireesquirez committed:
Final changes to "Contact Us"

voltaireesquirez committed:
Final changes to DG

voltaireesquirez committed:
Final changes to UG

voltaireesquirez committed:
Final changes to docs

voltaireesquirez committed:
Final changes to PPP

Hexinyiyi committed:
Merge pull request #286 from YezhongZ/refactor

Refactor ConnectToGoogleCal, FormatUtil

voltaireesquirez committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main into update_documentation

# Conflicts:
#	docs/UserGuide.adoc

voltaireesquirez committed:
Fix trailing whitespaces in docs

zylonc committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main into documentation

voltaireesquirez committed:
Final changes to docs

zylonc committed:
Update Delete implementation

jiacheng-panda committed:
Merge pull request #288 from ChengYuuu/documentation

Documentation

voltaireesquirez committed:
docs/: Update diagrams and images

voltaireesquirez committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main into update_documentation

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc

voltaireesquirez committed:
docs/: Fix remaining diagrams and images

voltaireesquirez committed:
docs/stylesheets/asciidoctor.css: Add a line to remove
remove weblinks from hyperlinks when converting
ASCII to PDF

voltaireesquirez committed:
Merge pull request #287 from CS2103-AY1819S1-F11-1/update_documentation

Some final changes to documentation

zylonc committed:
DeveloperGuide.adoc: Update Delete implementation

zylonc committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F11-1/main into documentation

Hexinyiyi committed:
update developer guide

Hexinyiyi committed:
conclude merging with org master branch

Hexinyiyi committed:
modification on developer guid

YipZong committed:
Merge pull request #291 from Hexinyiyi/master

Update developer guide

YipZong committed:
Merge pull request #290 from ChengYuuu/documentation

DeveloperGuide.adoc: Update Delete implementation

YipZong committed:
Merge

YipZong committed:
updatePPP

YipZong committed:
upload1

YipZong committed:
Merge branch 'master' into updateDocu

YipZong committed:
update 2

Hexinyiyi committed:
update ppp

Hexinyiyi committed:
Merge branch 'org'

YipZong committed:
update2

Hexinyiyi committed:
update PPP

YipZong committed:
update3

Hexinyiyi committed:
delete repeated parts in user guide

YipZong committed:
update PPP

zylonc committed:
Merge pull request #294 from YezhongZ/updateDocu

Update docu

zylonc committed:
Merge pull request #297 from Hexinyiyi/master

Update ppp

YipZong committed:
upload pic

zylonc committed:
Merge pull request #298 from YezhongZ/updateDocu

upload pic
